src/iceberg/manifest_entry.h

```diff
@@ -0,0 +1,256 @@
+  std::optional<int64_t> first_row_id;
+  /// Field id: 143
+  /// Fully qualified location (URI with FS scheme) of a data file that all deletes
+  /// reference.
+  ///
+  /// Position delete metadata can use referenced_data_file when all deletes tracked by
+  /// the entry are in a single data file. Setting the referenced file is required for
+  /// deletion vectors.
+  std::optional<std::string> referenced_data_file;
+  /// Field id: 144
+  /// The offset in the file where the content starts.
+  ///
+  /// The content_offset and content_size_in_bytes fields are used to reference a specific
+  /// blob for direct access to a deletion vector. For deletion vectors, these values are
+  /// required and must exactly match the offset and length stored in the Puffin footer
+  /// for the deletion vector blob.
+  std::optional<int64_t> content_offset;
+  /// Field id: 145
+  /// The length of a referenced content stored in the file; required if content_offset is
+  /// present
+  std::optional<int64_t> content_size_in_bytes;
+
+  static const SchemaField CONTENT;
+  static const SchemaField FILE_PATH;
+  static const SchemaField FILE_FORMAT;
+  static const SchemaField RECORD_COUNT;
+  static const SchemaField FILE_SIZE;
+  static const SchemaField COLUMN_SIZES;
+  static const SchemaField VALUE_COUNTS;
+  static const SchemaField NULL_VALUE_COUNTS;
+  static const SchemaField NAN_VALUE_COUNTS;
+  static const SchemaField LOWER_BOUNDS;
+  static const SchemaField UPPER_BOUNDS;
+  static const SchemaField KEY_METADATA;
+  static const SchemaField SPLIT_OFFSETS;
+  static const SchemaField EQUALITY_IDS;
+  static const SchemaField SORT_ORDER_ID;
+  static const SchemaField FIRST_ROW_ID;
+  static const SchemaField REFERENCED_DATA_FILE;
+  static const SchemaField CONTENT_OFFSET;
+  static const SchemaField CONTENT_SIZE;
+
+  static StructType GetType(StructType partition_type);
+};
+
+/// \brief A manifest is an immutable Avro file that lists data files or delete files,
+/// along with each file's partition data tuple, metrics, and tracking information.
+
+/// \brief The schema of a manifest file
+struct ICEBERG_EXPORT ManifestEntry {
+  /// Field id: 0
+  /// Used to track additions and deletions. Deletes are informational only and not used
+  /// in scans.
+  ManifestStatus status;
+  /// Field id: 1
+  /// Snapshot id where the file was added, or deleted if status is 2. Inherited when
+  /// null.
+  std::optional<int64_t> snapshot_id;
+  /// Field id: 3
+  /// Data sequence number of the file. Inherited when null and status is 1 (added).
+  std::optional<int64_t> sequence_number;
+  /// Field id: 4
+  /// File sequence number indicating when the file was added. Inherited when null and
+  /// status is 1 (added).
+  std::optional<int64_t> file_sequence_number;
+  /// Field id: 2
+  /// File path, partition tuple, metrics, ...
+  DataFile data_file;
```

Comment:
Should we use `std::unique_ptr` to avoid copy or `std::shared_ptr` to make the copy trivial?